test/integration/integration_test.go

```diff
@@ -0,0 +1,112 @@
+	integrationTestAppName = "integration-test-app"
+)
+
+func init() {
+	RegisterFlags(flag.CommandLine)
+}
+
+type LocalTestContextType struct {
+	AssetsDir string
+}
+
+var LocalTestContext LocalTestContextType
+
+func RegisterFlags(flags *flag.FlagSet) {
+	// Flags also used by the upstream test framework
+	flags.StringVar(&framework.TestContext.KubeConfig, clientcmd.RecommendedConfigPathFlag, os.Getenv(clientcmd.RecommendedConfigPathEnvVar), "Path to kubeconfig containing embedded authinfo.")
+	flags.StringVar(&framework.TestContext.KubeContext, clientcmd.FlagContext, "", "kubeconfig context to use/override. If unset, will use value from 'current-context'")
+	flags.StringVar(&framework.TestContext.KubectlPath, "kubectl-path", "kubectl", "The kubectl binary to use. For development, you might use 'cluster/kubectl.sh' here.")
+	flags.StringVar(&framework.TestContext.Host, "host", "", fmt.Sprintf("The host, or apiserver, to connect to. Will default to %s if this argument and --kubeconfig are not set", defaultHost))
+	flag.StringVar(&framework.TestContext.Provider, "provider", "", "The name of the Kubernetes provider (gce, gke, local, skeleton (the fallback if not set), etc.)")
+
+	// Required to prevent metrics from being annoyingly dumped to stdout
+	flag.StringVar(&framework.TestContext.GatherMetricsAfterTest, "gather-metrics-at-teardown", "false", "If set to 'true' framework will gather metrics from all components after each test. If set to 'master' only master component metrics would be gathered.")
+	flag.StringVar(&framework.TestContext.GatherKubeSystemResourceUsageData, "gather-resource-usage", "false", "If set to 'true' or 'all' framework will be monitoring resource usage of system all add-ons in (some) e2e tests, if set to 'master' framework will be monitoring master node only, if set to 'none' of 'false' monitoring will be turned off.")
+
+	// Custom flags
+	flags.StringVar(&LocalTestContext.AssetsDir, "assets", "assets", "The directory that holds assets used by the integration test.")
+
+	// Configure ginkgo as done by framework.RegisterCommonFlags
+	// Turn on verbose by default to get spec names
+	config.DefaultReporterConfig.Verbose = true
+
+	// Turn on EmitSpecProgress to get spec progress (especially on interrupt)
+	config.GinkgoConfig.EmitSpecProgress = true
+
+	// Randomize specs as well as suites
+	config.GinkgoConfig.RandomizeAllSpecs = true
+}
+
+func TestIntegration(t *testing.T) {
+	flag.Parse()
+	framework.AfterReadingAllFlags(&framework.TestContext)
+	RegisterFailHandler(ginkgo.Fail)
+	ginkgo.RunSpecs(t, "Amazon VPC CNI Integration Tests")
+}
+
+var _ = ginkgo.BeforeSuite(func() {
+	fmt.Printf("Using KUBECONFIG=\"%s\"\n", framework.TestContext.KubeConfig)
+})
+
+var _ = ginkgo.Describe("[cni-integration]", func() {
+	var f *framework.Framework
+	f = framework.NewDefaultFramework("cni-integration")
+
+	ginkgo.It("should enable pod-pod communication", func() {
+		serverPod := newTestPod()
```

Comment:
I think you need to clean up this resource, no? Otherwise, pods will continue to build up on the test cluster until the cluster is destroyed.

Reply:
Added a clean up.

Reply:
Prefer to set DeleteNamespaceOnFailure https://github.com/kubernetes/kubernetes/blob/cf06a2efd6ec950dc6707b57cd55cb4f50d29f33/test/e2e/framework/test_context.go#L278 true.

Reply:
> Prefer to set DeleteNamespaceOnFailure https://github.com/kubernetes/kubernetes/blob/cf06a2efd6ec950dc6707b57cd55cb4f50d29f33/test/e2e/framework/test_context.go#L278 true.

The problem with that is that you cannot diagnose something if a failure occurs.